yt/yt/server/node/exec_node/job_environment.cpp

```diff
@@ -1102,6 +1102,11 @@ class TCriJobEnvironment
         }
         spec->Credentials.Gid = ::getgid();
 
+        if (!config->GpuIndexes.empty()) {
+            spec->Environment["NVIDIA_DRIVER_CAPABILITIES"] = "all";
+            spec->Environment["NVIDIA_VISIBLE_DEVICES"] = JoinSeq(",", config->GpuIndexes);
```

Comment:
It would be nice to add a test checking that NVIDIA_VISIBLE_DEVICES passed by user in the operation spec will not grant them permission to use GPUs. However it may require some complex testing machinery, so let's leave it as is in case if it is hard.

Reply:
Based on code and manual testing, user's environment variables are not applied to job proxy environment
Automatic testing to verify this would be nice